### DISCUSSION TITLE
Constantly reconnecting, instead of steady connection (NordVPN)

### DISCUSSION THREAD

**Discussion by iluvatyr** [qdm12/gluetun#1082] [Q&A] [answered]
As seen in the complete logs here by looking for "INFO [vpn] stopping", every now and then, the VPN gets unhealthy and then (fortunately) restarts. 
Wouldnt it be possible to have a connection without reconnecting all the time?
Can someone see why it is doing that? Is it maybe something that nordvpn does and cannot be changed? 

[Gluetun Logs](https://termbin.com/cb1i)

**Comment by andry81** [comment]
Just curious. I've saw that behaviour with another VPN provider software on Windows and believe it does not release connection resources properly. So, VPN does reconnect on connections exhausting or hit the limit.
The solution is might be to forcibly close all unused or zombie connections.

Here is example of a solution in Windows:
https://superuser.com/questions/67039/how-to-kill-windows-zombie-tcp-connections/70888#70888

> CLOSE_WAIT means that the connection was closed on the other end.
> 
> Evidently, beyondtv doesn't detect this condition and continues to send data to the application on the other end. The other end can't send anything back over this connection, since it has closed its end of the connection.
> 
> The solution is to set the [TcpTimedWaitDelay](http://technet.microsoft.com/en-us/library/cc938217.aspx) entry in
> 
>     HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\tcpip\Parameters
> 
>     This entry determines the time that must elapse before TCP can release a closed connection and reuse its resources. This interval between closure and release is known as the TIME_WAIT state or 2MSL state. During this time, the connection can be reopened at much less cost to the client and server than establishing a new connection.
> 
> By default on my machine, this contains the value of -1, which I take to mean that closed connections are never released, which is exactly the behavior that you're observing.
> 
> I suggest that you set the value of this entry in the allowed range of 30–300 seconds. I suppose that 300 seconds = 5 minutes is entirely sufficient for your case, where it takes 40 minutes to freeze out your computer.

**Reply by qdm12** [reply]
@andry81 thanks for the suggestion. But this is related to the healthcheck failing triggering a vpn restart, not about it failing to reconnect after a failed healthcheck.

On that healthcheck, it's custom code to detect internet connectivity, see https://github.com/qdm12/gluetun/discussions/1082#discussioncomment-3273484

**Comment by qdm12** [comment, marked as answer]
Please read https://github.com/qdm12/gluetun/wiki/Healthcheck

You can adjust these options: https://github.com/qdm12/gluetun/wiki/Health-options

The stop and restart is configured to be rather aggressive, so if the connection to cloudflare.com:443 fails 5 times in a row within 6s, the VPN gets restarted.

**Reply by iluvatyr** [reply]
This solved it!  I set it to: 

                "HEALTH_VPN_DURATION_ADDITION=10s",
                "HEALTH_VPN_DURATION_INITIAL=25s",

and now it does not stop and reconnect like it did before.
Thank you a lot for the fast response!

**Reply by qdm12** [reply]
Feel free to try figuring out the lowest stable values if you want, and I can update the default values to these, since 6s/3s might be a bit too aggressive in the end.
